src/main/java/dev/morling/onebrc/CalculateAverage_flippingbits.java

```diff
@@ -0,0 +1,211 @@
+        var fileSize = file.length();
+        // Split file into chunks, so we can work around the size limitation of channels
+        var chunks = (int) (fileSize / CHUNK_SIZE);
+
+        long[][] chunkBoundaries = new long[chunks + 1][2];
+        var endPointer = 0L;
+
+        for (var i = 0; i <= chunks; i++) {
+            // Start of chunk
+            chunkBoundaries[i][0] = Math.min(Math.max(endPointer, i * CHUNK_SIZE), fileSize);
+
+            // Seek end of chunk, limited by the end of the file
+            file.seek(Math.min(chunkBoundaries[i][0] + CHUNK_SIZE - 1, fileSize));
+
+            // Extend chunk until end of line or file
+            while (true) {
+                var character = file.read();
+                if (character == '\n' || character == -1) {
+                    break;
+                }
+            }
+
+            // End of chunk
+            endPointer = file.getFilePointer();
+            chunkBoundaries[i][1] = endPointer;
+        }
+
+        return chunkBoundaries;
+    }
+
+    private static Map<String, PartitionAggregate> processChunk(long startOfChunk, long endOfChunk)
+            throws IOException {
+        Map<String, PartitionAggregate> stationAggregates = new HashMap<>(10_000);
+        byte[] byteChunk = new byte[(int) (endOfChunk - startOfChunk)];
+        try (var file = new RandomAccessFile(FILE, "r")) {
+            file.seek(startOfChunk);
+            file.read(byteChunk);
+            var i = 0;
+            while (i < byteChunk.length) {
+                final var startPosStation = i;
+
+                // read station name
+                while (byteChunk[i] != ';') {
+                    i++;
+                }
+                var station = new String(Arrays.copyOfRange(byteChunk, startPosStation, i));
+                i++;
+
+                // read measurement
+                final var startPosMeasurement = i;
+                while (byteChunk[i] != '\n') {
+                    i++;
+                }
+
+                var measurement = Arrays.copyOfRange(byteChunk, startPosMeasurement, i);
+                var aggregate = stationAggregates.getOrDefault(station, new PartitionAggregate());
+                aggregate.addMeasurementAndComputeAggregate(measurement);
+                stationAggregates.put(station, aggregate);
+                i++;
+            }
+            stationAggregates.values().forEach(PartitionAggregate::aggregateRemainingMeasurements);
+        }
+
+        return stationAggregates;
+    }
+
+    private static class PartitionAggregate {
+        final short[] lane = new short[SIMD_LANE_LENGTH * 2];
+        // Assume that we do not have more than Integer.MAX_VALUE measurements for the same station per partition
+        int count = 0;
+        long sum = 0;
+        short min = Short.MAX_VALUE;
+        short max = Short.MIN_VALUE;
+
+        public void addMeasurementAndComputeAggregate(byte[] measurementBytes) {
```

Comment:
Nice, I think it's the first contribution I'm seeing to use vector ops for the book keeping here.